roadmap/implementers-guide/src/node/collators/collation-generation.md

```diff
@@ -25,6 +25,13 @@ Collation generation for Parachains currently works in the following way:
   - No more than one initialization message should ever be sent to the collation
     generation subsystem.
   - Sent by a collator to initialize this subsystem.
+- `CollationGenerationMessage::SubmitCollation`
+  - If the subsystem isn't initialized or the relay-parent is too old to be relevant, ignore the message.
```

Comment:
I don't see an explicit check of relay parent age, though I'm guessing some of our relay runtime api requests would fail when the relay parent is too old. Is the cutoff 6 sessions?

Perhaps `request_validators` in `handle_submit_collation` would be the first thing to fail?

Reply:
State is discarded for finalized blocks, at the very least. 
If the relay parent doesn't fit into allowed ancestry, collator protocol will reject the candidate.

Reply:
This won't work for any blocks which have pruned state. The collation generation subsystem will handle requests that don't have too old a state, but then the collator protocol subsystem will ignore it if it's not one of the allowed relay parents.